Sources/App/Controllers/User/Auth/FacebookController.swift

```diff
@@ -0,0 +1,109 @@
+import HTTP
+import Vapor
+import Fluent
+
+final class FacebookController {
+
+  private let drop: Droplet
+  private let config: Config
+  private let photoController: PhotoConroller
+  private let fb = Social.FB.self
+
+  init(drop: Droplet) {
+    self.drop = drop
+    self.config = drop.config
+    self.photoController = PhotoConroller(drop: self.drop)
+  }
+
+  func createOrUpdateUserProfile(use authorizateCode: String) throws -> User {
```

Comment:
Need `authorizationCode` here. And maybe use `with` instead of `use`? To make in more English )) User profile with authorization code. The same is with `getUserProfile` below.

Reply:
yea, you right, fixed! Thanks!